### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by sosauce2**
- Updated Dependencies
- Updated to version catalog
- Fixed some deprecations
- Simplified some logics (gettings current version, opening links)
- Updated local songs to show their embedded art instead of the mediastore's default (
before: https://i.ibb.co/L0nCrpr/Screenshot-20240602-011454-Vibe-You.png
after: https://i.ibb.co/1sgnFPp/Screenshot-20240602-011427-Vibe-You-Debug.png)

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -12,7 +13,7 @@ android {
 
     defaultConfig {
         applicationId = "app.suhasdissa.vibeyou"
-        minSdk = 24
+        minSdk = 26
```
Comment (SuhasDissa) [inline app/build.gradle.kts:16]: What is the reason for this minSdk bump?

File: app/src/main/java/app/suhasdissa/vibeyou/presentation/components/SongCard.kt [hunk 2]
```diff
@@ -69,7 +84,7 @@ fun SongCard(
                 .padding(8.dp)
                 .aspectRatio(1f)
                 .clip(RoundedCornerShape(8.dp)),
-            model = song.thumbnailUri,
+            model = art,
```
Comment (SuhasDissa) [inline app/src/main/java/app/suhasdissa/vibeyou/presentation/components/SongCard.kt:87]: Album art of piped songs doesn't load with this method

File: app/build.gradle.kts (hunk 1 above)
Comment (sosauce2) [reply app/build.gradle.kts:16]: Was needed for the music art change

File: app/src/main/java/app/suhasdissa/vibeyou/presentation/components/SongCard.kt (hunk 2 above)
Comment (sosauce2) [reply app/src/main/java/app/suhasdissa/vibeyou/presentation/components/SongCard.kt:87]: Oopsies sorry! Didn't test that..

**Comment by sosauce2** [comment]
Fixed piped album art not displaying by having a separate LocalSongCard

**Comment by SuhasDissa** [comment]
Hey there,

I've reviewed this pull request and discussed these changes with other members.

First and foremost, I want to mention that we prefer pull requests that make smaller, incremental changes. Large pull requests with many changes at once make it difficult to review thoroughly and selectively accept specific changes.

We've decided to approve your changes to the `build.gradle.kts` files. However, we can't approve the `minSdk` version bump at this time.

Additionally, we believe your solution for loading song thumbnails from media metadata might have a negative impact on performance. Therefore, we won't be accepting that change.

Please update your pull request accordingly, or create a new pull request with the approved changes.

Thank you.

**Comment by sosauce2** [comment]
Hello ! Thanks for the feedback, will be more carefull. I will clise this PR 😄

**Pull request closed without merging**